src/ccl_power.c

```diff
@@ -406,194 +443,125 @@ static void ccl_cosmology_compute_power_class(ccl_cosmology * cosmo, int * statu
   if (*status != CCL_ERROR_CLASS)
     ccl_run_class(cosmo, &fc,&pr,&ba,&th,&pt,&tr,&pm,&sp,&nl,&le,&op,init_arr,status);
 
-  if (*status == CCL_ERROR_CLASS) {
-    //printed error message while running CLASS
-    ccl_free_class_structs(cosmo, &ba,&th,&pt,&tr,&pm,&sp,&nl,&le,init_arr,status);
-    
-    // free the parser
-    parser_free(&fc);
-    return;
+  if (*status == 0) {
+    if (parser_free(&fc)== _FAILURE_) {
+      *status = CCL_ERROR_CLASS;
+      ccl_cosmology_set_status_message(cosmo, "ccl_power.c: ccl_cosmology_compute_power_class(): Error freeing CLASS parser\n");
+    }
```

Comment:
I don't see the place where the parser is freed if status is non-zero. Did we reintroduce this memory leak?

Reply:
Right, the whole file has been refactored (that's why I didn't push to solving this earlier, since I thought this would be fixed when this got merged). This is now done in https://github.com/LSSTDESC/CCL/blob/3a3f5e9a2194465312e03b7782134ea0876eb07c/src/ccl_power.c#L446